e2e/utils.go

```diff
@@ -82,6 +85,133 @@ var (
 	clusterID        string
 )
 
+type cephfsFilesystem struct {
+	Name           string   `json:"name"`
+	MetadataPool   string   `json:"metadata_pool"`
+	MetadataPoolID int      `json:"metadata_pool_id"`
+	DataPoolIDs    []int    `json:"data_pool_ids"`
+	DataPools      []string `json:"data_pools"`
+}
+
+// listCephFSFileSystems list CephFS filesystems in json format.
+func listCephFSFileSystems(f *framework.Framework) ([]cephfsFilesystem, error) {
+	var fsList []cephfsFilesystem
+
+	stdout, stdErr, err := execCommandInToolBoxPod(
+		f,
+		"ceph fs ls --format=json",
+		rookNamespace)
+	if err != nil {
+		return fsList, err
+	}
+	if stdErr != "" {
+		return fsList, fmt.Errorf("error listing fs %v", stdErr)
+	}
+
+	err = json.Unmarshal([]byte(stdout), &fsList)
+	if err != nil {
+		return fsList, err
+	}
+
+	return fsList, nil
+}
+
+// getCephFSPoolName get CephFS pool name from filesystem name.
+func getCephFSPoolName(f *framework.Framework, filesystem string) (string, error) {
+	fsList, err := listCephFSFileSystems(f)
+	if err != nil {
+		return "", err
+	}
+	for _, v := range fsList {
+		if v.Name != filesystem {
+			continue
+		}
+
+		return v.MetadataPool, nil
+	}
+
+	return "", os.ErrNotExist
```

Comment:
`os.ErrNotExist` is not a proper error message for pools as its for files. use fmt.Errorf here?